### PR TITLE
SelectHowToPay for Soylent Seedling Systems.

### PR DESCRIPTION
**Pull request by kberg**
I'm breaking this into three or four pieces because without tests this
touches 25 files. That's how complicated our payment structure is; you
have to go through quite a lot of layers. I'd like to simplify this just
a little more before adding one more of these. No need for all the
hardcoding.

### CODE REVIEW

File: src/client/components/SelectHowToPay.vue [hunk 1]
```diff
@@ -275,6 +284,14 @@ export default Vue.extend({
       <Button type="max" @click="setMaxValue('heat')" title="MAX" />
     </div>
 
+    <div class="payments_type input-group">
```
Comment (bafolts) [inline src/client/components/SelectHowToPay.vue:287]: Will this always show for a user even if paying with seeds is irrelevant for their game options?

File: src/client/components/SelectHowToPayForProjectCard.vue [hunk 2]
```diff
@@ -234,6 +241,19 @@ export default Vue.extend({
...
     },
+    canUseSeeds() {
+      // FYI Seed Resources are limited to the Soylent Seedling Systems corp card, which allows spending its
+      // resources for plant cards and the standard greenery project.
+      if (this.card !== undefined && (this.playerinput.seeds ?? 0) > 0) {
+        if (this.tags.includes(Tags.PLANT)) {
+          return true;
+        }
+        if (this.card.name === CardName.GREENERY_STANDARD_PROJECT) {
+          return true;
```
Comment (bafolts) [inline src/client/components/SelectHowToPayForProjectCard.vue:252]: Well this worked out well. Benefit of having standard projects as cards.

Comment (kberg) [reply src/client/components/SelectHowToPayForProjectCard.vue:252]: Didn't it?

File: src/client/components/SelectHowToPay.vue (hunk 1 above)
Comment (kberg) [reply src/client/components/SelectHowToPay.vue:287]: Whoops! It will. I'll fix that.

File: src/client/components/SelectHowToPay.vue [hunk 3]
```diff
@@ -276,6 +285,14 @@ export default Vue.extend({
...
 
     <div class="payments_type input-group">
+      <i class="resource_icon resource_icon--seeds payments_type_icon" :title="$t('Pay by Seeds')"></i>
+      <Button type="minus" @click="reduceValue('seeds', 1)" />
+      <input class="form-input form-inline payments_input" v-model.number="seeds" />
+      <Button type="plus" @click="addValue('seeds', 1)" />
+      <Button type="max" @click="setMaxValue('seeds')" title="MAX" />
+    </div>
+
+    <div class="payments_type input-group" v-if="playerinput.canUseSeeds">
```
Comment (bafolts) [inline src/client/components/SelectHowToPay.vue:295]: I think this should be on the div for paying with divs and not on this div for paying with mega credits.

Comment (kberg) [reply src/client/components/SelectHowToPay.vue:295]: So sloppy! That's why I added a test. SRY SRY SRY

Comment (kberg) [reply src/client/components/SelectHowToPay.vue]: OK tests pass now.  Time is valuable sorry for wasting yours! :D

File: tests/components/PaymentTester.ts [hunk 4]
```diff
@@ -41,30 +41,7 @@ export class PaymentTester {
...
-      vmVal = this.model.microbes;
-      break;
-    case 'megaCredits':
-      vmVal = this.model.megaCredits;
-      break;
-    case 'science':
-      vmVal = this.model.science;
-      break;
-    }
+    const vmVal = this.model[type];
```
Comment (bafolts) [inline tests/components/PaymentTester.ts:44]: This is a great change. 👏

Comment (bafolts) [reply tests/components/PaymentTester.ts:44]: I see you.

**Review by bafolts — APPROVED** [review]

File: tests/components/PaymentTester.ts (hunk 4 above)
Comment (kberg) [reply tests/components/PaymentTester.ts:44]: I keep trying to simplify SHTP and once in a while it works.

**Pull request merged**